build.gradle

```diff
@@ -20,6 +21,9 @@ dependencies {
     compile 'org.bitbucket.b_c:jose4j:0.4.4'
     compile 'org.bouncycastle:bcprov-jdk15on:1.52'
     compile 'org.bouncycastle:bcpkix-jdk15on:1.52'
+    compile 'com.google.code.gson:gson:1.7.2'
```

Comment:
Given we are new to the codebase, we might need some direction here. Hope you can help us!

All the existing code I found using `JsonObject.readFrom(...)` seem to be directly tied to the extension of `parseJSONMember(JsonObject.member)` in the `BoxResource.Info` hierarchy. An example of that is:

```@Override
        protected void parseJSONMember(JsonObject.Member member) {
            super.parseJSONMember(member);

            try {
                JsonValue value = member.getValue();
                String memberName = member.getName();
                if (memberName.equals("sequence_id")) {
                    this.sequenceID = value.asString();
                } else if (memberName.equals("etag")) {
                    this.etag = value.asString();
                } else if (memberName.equals("name")) {
                    this.name = value.asString();
                } else if (memberName.equals("created_at")) {
                    this.createdAt = BoxDateFormat.parse(value.asString());
                } else if (memberName.equals("modified_at")) {
                    this.modifiedAt = BoxDateFormat.parse(value.asString());
                } else if (memberName.equals("description")) {
                    this.description = value.asString();
                } else if (memberName.equals("size")) {
                    this.size = Double.valueOf(value.toString()).longValue();
                } else if (memberName.equals("trashed_at")) {
                    this.trashedAt = BoxDateFormat.parse(value.asString());
                } else if (memberName.equals("purged_at")) {
                    this.purgedAt = BoxDateFormat.parse(value.asString());
                } else if (memberName.equals("content_created_at")) {
                    this.contentCreatedAt = BoxDateFormat.parse(value.asString());
                } else if (memberName.equals("content_modified_at")) {
                    this.contentModifiedAt = BoxDateFormat.parse(value.asString());
                } else if (memberName.equals("path_collection")) {
                    this.pathCollection = this.parsePathCollection(value.asObject());
                } else if (memberName.equals("created_by")) {
                    this.createdBy = this.parseUserInfo(value.asObject());
                } else if (memberName.equals("modified_by")) {
                    this.modifiedBy = this.parseUserInfo(value.asObject());
                } else if (memberName.equals("owned_by")) {
                    this.ownedBy = this.parseUserInfo(value.asObject());
                } else if (memberName.equals("shared_link")) {
                    if (this.sharedLink == null) {
                        this.setSharedLink(new BoxSharedLink(value.asObject()));
                    } else {
                        this.sharedLink.update(value.asObject());
                    }
                } else if (memberName.equals("tags")) {
                    this.tags = this.parseTags(value.asArray());
                } else if (memberName.equals("parent")) {
                    JsonObject jsonObject = value.asObject();
                    if (this.parent == null) {
                        String id = jsonObject.get("id").asString();
                        BoxFolder parentFolder = new BoxFolder(getAPI(), id);
                        this.parent = parentFolder.new Info(jsonObject);
                    } else {
                        this.parent.update(jsonObject);
                    }
                } else if (memberName.equals("item_status")) {
                    this.itemStatus = value.asString();
                }
            } catch (ParseException e) {
                assert false : "A ParseException indicates a bug in the SDK.";
            }
        }
```

I'm assuming there are simpler, less manual and error-prone ways to map from and to JSON, hopefully still keeping the lightweight dependencies that are already there. Where could I check for those?

Thanks!